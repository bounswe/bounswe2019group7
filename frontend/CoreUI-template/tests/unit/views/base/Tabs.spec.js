import Vue from 'vue'
import { mount, shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue'
import Tabs from '@/views/base/Tabs'

Vue.use(BootstrapVue)

describe('Tabs.vue', () => {
  it('has a name', () => {
    expect(Tabs.name).toMatch('tabs')
  })
  it('has a created hook', () => {
    expect(typeof Tabs.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Tabs.data).toMatch('function')
    const defaultData = Tabs.data()
    expect(defaultData.tabs).toEqual(["Calculator", "Shopping cart", "Charts"])
    expect(defaultData.tabIndex).toEqual([0, 0])
  })
  it('is Vue instance', () => {
    const wrapper = mount(Tabs)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Tabs', () => {
    const wrapper = mount(Tabs)
    expect(wrapper.is(Tabs)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Tabs)
    expect(wrapper.find('div.card-header > div').text()).toMatch('Tabs')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Tabs)
    expect(wrapper.element).toMatchSnapshot()
  })
})
