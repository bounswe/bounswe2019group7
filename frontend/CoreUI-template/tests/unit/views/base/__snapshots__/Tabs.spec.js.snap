// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs.vue renders correctly 1`] = `
<div
  class="animated fadeIn"
>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <b-card-stub
        bodytag="div"
        footertag="div"
        headertag="div"
        subtitle=""
        subtitletag="h6"
        subtitletextvariant="muted"
        tag="div"
        title=""
        titletag="h4"
      >
        <div
          slot="header"
        >
          
          Tabs
        
        </div>
         
        <b-tabs-stub
          tag="div"
        >
          <b-tab-stub
            active="true"
            buttonid=""
            href="#"
            tag="div"
            title="Home"
          >
            <br />
             1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
           
          <b-tab-stub
            buttonid=""
            href="#"
            tag="div"
            title="Profile"
          >
            <br />
            2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
           
          <b-tab-stub
            buttonid=""
            disabled="true"
            href="#"
            tag="div"
            title="Disabled"
          >
            <br />
            3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
        </b-tabs-stub>
      </b-card-stub>
    </b-col-stub>
     
    <b-col-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <b-card-stub
        bodytag="div"
        footertag="div"
        headertag="div"
        subtitle=""
        subtitletag="h6"
        subtitletextvariant="muted"
        tag="div"
        title=""
        titletag="h4"
      >
        <div
          slot="header"
        >
          
          Tabs with icons
        
        </div>
         
        <b-tabs-stub
          tag="div"
        >
          <b-tab-stub
            active="true"
            buttonid=""
            href="#"
            tag="div"
            title=""
          >
            <template>
              <i
                class="icon-calculator"
              />
            </template>
             
            <br />
             1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
           
          <b-tab-stub
            buttonid=""
            href="#"
            tag="div"
            title=""
          >
            <template>
              <i
                class="icon-basket-loaded"
              />
            </template>
             
            <br />
            2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
           
          <b-tab-stub
            buttonid=""
            href="#"
            tag="div"
            title=""
          >
            <template>
              <i
                class="icon-pie-chart"
              />
            </template>
             
            <br />
            3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
        </b-tabs-stub>
      </b-card-stub>
    </b-col-stub>
     
    <b-col-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <b-card-stub
        bodytag="div"
        footertag="div"
        headertag="div"
        subtitle=""
        subtitletag="h6"
        subtitletextvariant="muted"
        tag="div"
        title=""
        titletag="h4"
      >
        <div
          slot="header"
        >
          
          Tabs with icons
        
        </div>
         
        <b-tabs-stub
          tag="div"
          value="0"
        >
          <b-tab-stub
            active="true"
            buttonid=""
            href="#"
            tag="div"
            title=""
          >
            <template>
              <i
                class="icon-calculator"
              />
               Calculator
            
            </template>
             
            <br />
             1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
           
          <b-tab-stub
            buttonid=""
            href="#"
            tag="div"
            title=""
          >
            <template>
              <i
                class="icon-basket-loaded"
              />
               Shopping cart
            
            </template>
             
            <br />
            2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
           
          <b-tab-stub
            buttonid=""
            href="#"
            tag="div"
            title=""
          >
            <template>
              <i
                class="icon-pie-chart"
              />
               Charts
            
            </template>
             
            <br />
            3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
          </b-tab-stub>
        </b-tabs-stub>
      </b-card-stub>
    </b-col-stub>
     
    <b-col-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <b-card-stub
        bodytag="div"
        footertag="div"
        headertag="div"
        subtitle=""
        subtitletag="h6"
        subtitletextvariant="muted"
        tag="div"
        title=""
        titletag="h4"
      >
        <div
          slot="header"
        >
          
          Tabs vertical
        
        </div>
         
        <b-card-stub
          bodytag="div"
          footertag="div"
          headertag="div"
          nobody="true"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <b-tabs-stub
            card="true"
            navwrapperclass="w-40"
            pills="true"
            tag="div"
            value="0"
            vertical="true"
          >
            <b-tab-stub
              active="true"
              buttonid=""
              href="#"
              tag="div"
              title=""
            >
              <template>
                <i
                  class="icon-calculator"
                />
                 Calculator
            
              </template>
              
            1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
            </b-tab-stub>
             
            <b-tab-stub
              buttonid=""
              href="#"
              tag="div"
              title=""
            >
              <template>
                <i
                  class="icon-basket-loaded"
                />
                 Shopping cart
            
              </template>
              
            2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
            </b-tab-stub>
             
            <b-tab-stub
              buttonid=""
              href="#"
              tag="div"
              title=""
            >
              <template>
                <i
                  class="icon-pie-chart"
                />
                 Charts
            
              </template>
              
            3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
            et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
            officia deserunt mollit anim id est laborum.
          
            </b-tab-stub>
          </b-tabs-stub>
        </b-card-stub>
      </b-card-stub>
    </b-col-stub>
  </b-row-stub>
</div>
`;
